### PR TITLE
use mobility segments

### PR DESCRIPTION
**Pull request by SteffenHeu**
instead of storing every frame's mobilities in a new double buffer, the mobilities are stored as an array in memory with the raw data file. a new segment is only added if the mobility values are not matching any other segment.

### CODE REVIEW

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java [hunk 1]
```diff
@@ -129,23 +127,27 @@ public MobilityScan getMobilityScan(int num) {
...
+          getMobilities().length, originalMobilityScans.size()));
     }
     mobilityScanStorage = new MobilityScanStorage(getDataFile().getMemoryMapStorage(), this,
         originalMobilityScans, useAsMassList);
   }
 
   @Override
   public double getMobilityForMobilityScanNumber(int mobilityScanIndex) {
-    return mobilityBuffer.get(mobilityScanIndex);
+    return ((IMSRawDataFileImpl) (getDataFile())).getSegmentMobilities(
```
Comment (robinschmid) [inline src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java:141]: Should be interface IMSRawDataFile

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java:141]: i didnt want to expose this method by default, so it's just in the implementation.

Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java:141]: Either use IMSRawDataFileImpl then in Frame to enforce this - casting is unsafe. 
or pull this method up to the interface

File: src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java [hunk 2]
```diff
@@ -172,17 +174,11 @@ public PasefMsMsInfo getImsMsMsInfoForMobilityScan(int mobilityScanNumber) {
...
-  public DoubleBuffer setMobilities(double[] mobilities) {
-    if (mobilityScanStorage != null && (getNumberOfMobilityScans() != mobilities.length)) {
-      throw new IllegalArgumentException(String.format(
-          "Number of mobility values (%d) does not match number of mobility scans (%d).",
-          mobilities.length, getNumberOfMobilityScans()));
-    }
-    mobilityBuffer = StorageUtils.storeValuesToDoubleBuffer(getDataFile().getMemoryMapStorage(),
-        mobilities);
+  public int setMobilities(double[] mobilities) {
+    mobilitySegment = ((IMSRawDataFileImpl) getDataFile()).addMobilityValues(mobilities);
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/impl/SimpleFrame.java:180]: still impl

**Comment by robinschmid** [comment]
I checked everything again. Great work. 
Only the two comments left. And the workflow does not run - I will restart it and check later

**Pull request merged**